# Imports

In [2]:
import pandas as pd
import statsmodels.api as sm
import matplotlib.pyplot as plt
from ipynb.fs.defs.problem1 import estimate_capm


## Load in Fama French Portfolio Data

In [6]:
# Read the file, skipping header text rows and limiting to monthly data
ff = pd.read_csv("F-F_Research_Data_Factors 3.CSV", skiprows=3, nrows=1182)

# Rename the date column for clarity
ff.rename(columns={"Unnamed: 0": "Date"}, inplace=True)

# Convert Date to string to extract Year and Month
ff['Date'] = ff['Date'].astype(str)
ff['Year'] = ff['Date'].str[:4].astype(int)
ff['Month'] = ff['Date'].str[4:].astype(int)

# Pull the columns we want
ff = ff[['Year', 'Month', 'Mkt-RF', 'RF']]

# Show the updated DataFrame
print(ff.head())

   Year  Month  Mkt-RF    RF
0  1926      7    2.96  0.22
1  1926      8    2.64  0.25
2  1926      9    0.36  0.23
3  1926     10   -3.24  0.32
4  1926     11    2.53  0.31


## Load in Strategies Dataset

In [4]:
# Read the file
ss = pd.read_csv("ps1_strategies.csv")

# Convert Date to string to extract Year and Month
ss['date'] = ss['date'].astype(str)
ss['Year'] = ss['date'].str[:4].astype(int)
ss['Month'] = ss['date'].str[4:].astype(int)

# Pull the columns we want
ss = ss[['Year', 'Month', 'LBHA', 'LSA', 'TA', 'HV', 'LV', 'NA', 'LB', 'HB']]

# Show the updated DataFrame
print(ss.head())

   Year  Month      LBHA     LSA        TA        HV        LV        NA  \
0  1990      1  1.498262 -7.4575  1.679061 -7.271919  0.022091 -5.392944   
1  1990      2  3.642659  1.0545  0.205289 -0.986167  0.062055 -1.768405   
2  1990      3  1.737180  1.7385 -1.572688 -0.018665  0.341639 -0.333926   
3  1990      4  0.734520 -3.1920  2.474704 -3.294381  0.253568 -2.578905   
4  1990      5  1.298923  7.9990  0.754379  8.038877 -0.113650  1.337511   

         LB         HB  
0 -1.353457 -22.772632  
1 -2.118514   5.151408  
2  1.452434   4.480134  
3  2.123740 -10.101798  
4 -1.555230  26.259080  


# Part (a)

In [7]:
results_lbha = estimate_capm(ss, ff, 'LBHA')

print("CAPM Results for LBHA Strategy")
print(f"Alpha: {results_lbha['alpha']:.4f}")
print(f"Beta: {results_lbha['beta']:.4f}")
print(results_lbha['summary'])

CAPM Results for LBHA Strategy
Alpha: 0.2729
Beta: 0.0067
                            OLS Regression Results                            
Dep. Variable:                  RP-RF   R-squared:                       0.000
Model:                            OLS   Adj. R-squared:                 -0.002
Method:                 Least Squares   F-statistic:                   0.08125
Date:                Thu, 10 Apr 2025   Prob (F-statistic):              0.776
Time:                        00:42:27   Log-Likelihood:                -853.10
No. Observations:                 397   AIC:                             1710.
Df Residuals:                     395   BIC:                             1718.
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------

Since alpha (0.2729) > 0 and p-value (0.01) is small (< 0.05), we can say with some confidence that LBHA has positive risk-adjusted outperformance; so, the LBHA strategy **beats the market**.